pkg/services/nmap.go

```diff
@@ -184,21 +178,18 @@ func handleScanErrors(ctx context.Context, err error, warnings []string, targetC
 		slog.Default().Warn("Nmap raised warnings:", slog.Any("nmap_warnings", warnings))
 	}
 	if err != nil {
-		if ctx.Err() == context.DeadlineExceeded {
-			return fmt.Errorf("scan timeout after %d seconds", 240*targetCount)
+		if errors.Is(ctx.Err(), context.Canceled) {
+			return fmt.Errorf("nmap service has been cancelled")
+		}
+		if errors.Is(ctx.Err(), context.DeadlineExceeded) {
+			return fmt.Errorf("context timeout after %d seconds", 1800*targetCount)
```

Comment:
Though this is right, it is kind of hardcoded and not taking into account our set timeout (think what happens if the developer increases the timeout to 1900 seconds, for example). As a simple fix, we can:

```suggestion
			return errors.New("context timed out")
```

Or we could pass the `scanTimeout` as a parameter to this func. I'll leave it up to you 😉